plugins/wm-actions/wm-actions.cpp

```diff
@@ -267,6 +268,38 @@ class wayfire_wm_actions_t : public wf::plugin_interface_t
         return true;
     };
 
+    wf::activator_callback on_send_to_back = [=] (auto ev) -> bool
+    {
+        return execute_for_selected_view(ev.source, [] (wayfire_view view)
+        {
+            auto ws    = view->get_output()->workspace->get_current_workspace();
+            auto views =
+                view->get_output()->workspace->get_views_on_workspace(ws,
+                    wf::LAYER_WORKSPACE);
+            wayfire_view bottom_view = views[views.size() - 1];
+            if (view != bottom_view)
+            {
+                // Move view to bottom by stacking it below the bottom-most view
+                view->get_output()->workspace->restack_below(view, bottom_view);
+
+                // Change focus to the last focused view on this workspace
```

Comment:
Sorry for the late suggestion, but this is mostly implemented already with this function https://github.com/WayfireWM/wayfire/blob/master/src/api/wayfire/output.hpp#L197

There's a catch though - you should check that there is at least one other view on the workspace, otherwise, `refocus()` will set focus to NULL. But I think that a simple check like that is better than reinventing `refocus()`

Reply:
```
  285                 // Change focus to the last focused view on this workspace
  286                 if (views.size() > 1) {
  287                     view->get_output()->refocus(view, wf::LAYER_WORKSPACE);
  288                 }
```
This isn't quite working. If I have three views open: A, B, C (which appear in scale in that order, and I keep triggering the send_to_back, I get:
1. A goes to bottom, B is focused and is the top-most view
2. B goes to bottom, C is the top-most view, but A is focused
Before, I could keep triggering it and I could loop through all three windows in the same order forever, which is the ideal behaviour really. Certainly, it's not acceptable that the top-most view isn't the one focused.

Reply:
Figured out the problem, don't quite understand it. This change in refocus makes it work:
```
@@ -92,8 +92,9 @@ void wf::output_impl_t::refocus(wayfire_view skip_view, uint32_t layers)
         bool visible_1 = view_on_current_ws(view1);
         bool visible_2 = view_on_current_ws(view2);

-        return std::make_tuple(visible_1, view1->last_focus_timestamp) <
-        std::make_tuple(visible_2, view2->last_focus_timestamp);
+        return (visible_1 && view1->last_focus_timestamp) < (visible_2 && view2->last_focus_timestamp);
```
So, it's the tuple comparison which doesn't seem to be working properly. This "fix" however treats all non-visible views the same, not caring about their last_focus_timestamp.

Reply:
I've submitted a commit which doesn't use last_focus_timestamp at all and just uses the stacking order as it was before, this is working in my tests.

Reply:
Okay, I guess it makes sense that we don't follow last-focused order when we're changing the stacking order. I'm leaving this for you to decide, current version LGTM.